### PR TITLE
feat: add relayer fee settings

### PR DESCRIPTION
**Pull request by byte-bandit**
# Related Github tickets

- https://github.com/VolumeFi/paloma/issues/1710

# Background

> This change adds the concept of individual
> relayer fees that may be set by any active
> validator for any supported target chain.

# Testing completed

- [x] test coverage exists or has been added/updated
- [x] tested in a private testnet

# Breaking changes

- [x] I have checked my code for breaking changes
- [x] If there are breaking changes, there is a supporting migration.

### CODE REVIEW

File: x/treasury/client/cli/tx.go [hunk 1]
```diff
@@ -21,5 +25,69 @@ func GetTxCmd() *cobra.Command {
...
+	return cmd
+}
+
+func CmdUpsertRelayerFee() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "upsert-relayer-fee [chain-reference-id] [fee-multiplicator]",
+		Short: "Sets the relayer fee for the sender to the given multiplicator.",
+		Long: `
+    Sets the relayer fee for the sender to th given multiplicator value.
```
Comment (maharifu) [inline x/treasury/client/cli/tx.go:38]:
```suggestion
    Sets the relayer fee for the sender to the given multiplicator value.
```

File: x/treasury/client/cli/tx.go [hunk 2]
```diff
@@ -21,5 +25,69 @@ func GetTxCmd() *cobra.Command {
...
+			if err != nil {
+				return err
+			}
+
+			id := args[0]
+			fee := math.LegacyMustNewDecFromStr(args[1])
+
+			if fee.LT(math.LegacyZeroDec()) {
+				return fmt.Errorf("negative fee not allowed")
+			}
```
Comment (maharifu) [inline x/treasury/client/cli/tx.go:64]: Is this needed? We're not allowing values less than 1 below.

**Review by maharifu — CHANGES_REQUESTED** [review]
Nice one! Just a couple of nitpicks.

File: proto/palomachain/paloma/treasury/query.proto [hunk 3]
```diff
@@ -19,6 +21,11 @@ service Query {
   rpc QueryFees(QueryFeesRequest) returns (Fees) {
     option (google.api.http).get = "/palomachain/paloma/treasury/fees";
   }
+
+  // RelayerFee queries the fee settings of a validator participating in consensus.
+  rpc RelayerFee(QueryRelayerFeeRequest) returns (RelayerFeeSetting) {
+    option (google.api.http).get = "/palomachain/paloma/treasury/relayer-fee";
```
[the remark this answers is not in the document]
Comment (byte-bandit) [reply proto/palomachain/paloma/treasury/query.proto:27]: I don't know. While there's no clear answer, general consensus seems to shift towards using hyphens over underscores. It's mostly for SEO, but I've become more used to them now.

That being said, all other URIs are using underscore, so better to stick with one than mix.

**Review by maharifu — APPROVED** [review]
LGTM

**Pull request merged**